### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by shahzaman900**
### Within this branch we have:

- We went through the main.rb file and made sure its within the rules of YAGNI, KISS, and DRY
- We have followed SOLID principles, and refactured main.rb file

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job updating the main.rb file

## Required Changes ♻️
### Because there are no changes in the app.rb file I'll leave the comments below:
- [ ] You are not complying with the Single responsibility rule, please have a look at [the Best practices and Solid lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information. All requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.
below you can see the methods in the app.rb file which all have the issue mentioned above.

https://github.com/sheezu900/school_lib/blob/03d7f1fd2301810030cfa50540b6caf1180a29ca/app.rb#L26-L43
https://github.com/sheezu900/school_lib/blob/03d7f1fd2301810030cfa50540b6caf1180a29ca/app.rb#L45-L55
https://github.com/sheezu900/school_lib/blob/03d7f1fd2301810030cfa50540b6caf1180a29ca/app.rb#L57-L64
https://github.com/sheezu900/school_lib/blob/03d7f1fd2301810030cfa50540b6caf1180a29ca/app.rb#L66-L74
https://github.com/sheezu900/school_lib/blob/03d7f1fd2301810030cfa50540b6caf1180a29ca/app.rb#L100-L107
https://github.com/sheezu900/school_lib/blob/03d7f1fd2301810030cfa50540b6caf1180a29ca/app.rb#L109-L116
https://github.com/sheezu900/school_lib/blob/03d7f1fd2301810030cfa50540b6caf1180a29ca/app.rb#L118-L121
https://github.com/sheezu900/school_lib/blob/03d7f1fd2301810030cfa50540b6caf1180a29ca/app.rb#L123-L136

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by addod19** [comment]
## Hi TEAM, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because we can not use a new PR for the previous review request. We are advised to use the same PR for all review requests until they are approved and merged

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -1,136 +1,112 @@
...
+      create_student(age, name, has_parent_permission)
+    elsif type == 2
+      puts 'Enter specialization'
+      specialization = gets.chomp
+      create_teacher(age, specialization, name)
     else
-      puts 'Invalid option.'
+      puts 'Invalid input'
     end
   end
```
Comment (ghost) [inline app.rb]: - [ ] As stated earlier,  Your App class is responsible for both user interaction (getting input from the user) and managing the data (books, people, rentals). Ideally, the responsibility for user interaction should be separated from the data management logic. This follows the Single Responsibility Principle and makes the code more modular and maintainable. To achieve this, you can create separate classes or modules for user interaction (getting input and displaying messages).

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ `Main.rb`  file is clean 👍 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi Team,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ]  Kudos for working on refactor🤩
- [ ]  No Linters error ✅


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**